src/test/java/org/opensearch/ad/ml/EntityColdStarterTests.java

```diff
@@ -458,4 +509,258 @@ public void testEmptyDataRange() throws InterruptedException {
+                AnomalyDetectorSettings.THRESHOLD_MIN_PVALUE,
+                AnomalyDetectorSettings.THRESHOLD_MAX_RANK_ERROR,
+                AnomalyDetectorSettings.THRESHOLD_MAX_SCORE,
+                AnomalyDetectorSettings.THRESHOLD_NUM_LOGNORMAL_QUANTILES,
+                AnomalyDetectorSettings.THRESHOLD_DOWNSAMPLES,
+                AnomalyDetectorSettings.THRESHOLD_MAX_SAMPLES,
+                featureManager,
+                AnomalyDetectorSettings.HOURLY_MAINTENANCE,
+                AnomalyDetectorSettings.MAX_SMALL_STATES,
+                checkpoint,
+                settings,
+                seed
+            );
+
+            // create labelled data
+            MultiDimDataWithTime dataWithKeys = LabelledAnomalyGenerator
+                .getMultiDimData(
+                    dataSize + detector.getShingleSize() - 1,
+                    50,
+                    100,
+                    5,
+                    seed,
+                    baseDimension,
+                    false,
+                    trainTestSplit,
+                    delta,
+                    anomalyIndependent
+                );
+
+            long[] timestamps = dataWithKeys.timestampsMs;
+            double[][] data = dataWithKeys.data;
+            when(clock.millis()).thenReturn(timestamps[trainTestSplit - 1]);
+
+            doAnswer(invocation -> {
+                ActionListener<Entry<Optional<Long>, Optional<Long>>> listener = invocation.getArgument(2);
+                listener.onResponse(new SimpleImmutableEntry<>(Optional.of(timestamps[0]), Optional.of(timestamps[trainTestSplit - 1])));
+                return null;
+            }).when(searchFeatureDao).getEntityMinMaxDataTime(any(), any(), any());
+
+            doAnswer(invocation -> {
+                List<Entry<Long, Long>> ranges = invocation.getArgument(1);
+                List<Optional<double[]>> coldStartSamples = new ArrayList<>();
+
+                Collections.sort(ranges, new Comparator<Entry<Long, Long>>() {
+                    @Override
+                    public int compare(Entry<Long, Long> p1, Entry<Long, Long> p2) {
+                        return Long.compare(p1.getKey(), p2.getKey());
+                    }
+                });
+                for (int j = 0; j < ranges.size(); j++) {
+                    Entry<Long, Long> range = ranges.get(j);
+                    Long start = range.getKey();
+                    int valueIndex = searchInsert(timestamps, start);
+                    coldStartSamples.add(Optional.of(data[valueIndex]));
+                }
+
+                ActionListener<List<Optional<double[]>>> listener = invocation.getArgument(4);
+                listener.onResponse(coldStartSamples);
+                return null;
+            }).when(searchFeatureDao).getColdStartSamplesForPeriods(any(), any(), any(), anyBoolean(), any());
+
+            String entityNamez = entityName + z;
```

Comment:
I'm a little confused about this part, are we creating a new Entity model that doesn't exist with name "abcz" and then training the model with that one entity so we treat all the data points only for that one entity as we are only training one model? basically simulating HCAD logic with only one entity? Does that mean the categories given earlier when creating the detector and going through cold start don't matter?

Reply:
yeah, what you described is correct. I am simulate HCAD logic with only one entity. It takes a long time run all of these steps for one entity (1.0 is relatively fast, 1.1 takes 25+ minutes since we disabled bounding box caching). Yeah, the categories given earlier does not matter.

Reply:
was the reason we needed the other anomaly detector and entity just to get samples for that period?

Reply:
We need the detector object as EntityColdStarter.getEntityColdStartData requires a full detector and then access its properties.